lib/client/auth-code.js

```diff
@@ -37,6 +37,8 @@ module.exports = function (config) {
    *                             An error object is passed as first argument and the result as last.
    */
   function getToken(params, callback) {
+    params.client_id = config.clientID;
+    params.client_secret = config.clientSecret;
```

Comment:
I have a couple of questions here. It seems like the standard oauth-flow only requires the client_id on this cases (see https://tools.ietf.org/html/rfc6749#section-1.3). But it also says that this parameter is not always required, but only under certain conditions indicated in there.

Have you tried to only send the client_id without the secret?
I also have doubts regarding on what most of authorization servers would do in case of strange params (since this param is not always required), what would happen on cases where is not actually required?